server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader12to13.java

```diff
@@ -70,6 +89,89 @@ public void upgradeMetadata(ServerContext context) {
     deleteExternalCompactionFinalStates(context);
     deleteExternalCompactions(context);
     removeMetaDataBulkLoadFilter(context, MetadataTable.ID);
+    LOG.info("Removing compact columns from user tables");
+    removeCompactColumnsFromTable(context, MetadataTable.NAME);
+  }
+
+  private void removeCompactColumnsFromRootTabletMetadata(ServerContext context) {
+    var rootBase = ZooUtil.getRoot(context.getInstanceID()) + ZROOT_TABLET;
+
+    try {
+      var zrw = context.getZooReaderWriter();
+      Stat stat = new Stat();
+      byte[] rootData = zrw.getData(rootBase, stat);
+
+      String json = new String(rootData, UTF_8);
+
+      var rtm = new RootTabletMetadata(json);
+
+      ArrayList<Mutation> mutations = new ArrayList<>();
+      for (Map.Entry<Key,Value> entry : rtm.toKeyValues().entrySet()) {
+        var key = entry.getKey();
+        var row = key.getRow();
+        Preconditions.checkState(key.getColumnVisibilityData().length() == 0,
+            "Expected empty visibility, saw %s ", key.getColumnVisibilityData());
+        Mutation m = new Mutation(row);
+        m.putDelete(MetadataSchema.TabletsSection.ServerColumnFamily.NAME, COMPACT_QUAL);
+        mutations.add(m);
```

Comment:
This generating a mutation for every column in the tablet metadata, only want to process the compact col. Could do the following.

```suggestion
        if(COMPACT_COL.hasColumns(key)) {
          var row = key.getRow();
          Preconditions.checkState(key.getColumnVisibilityData().length() == 0,
                  "Expected empty visibility, saw %s ", key.getColumnVisibilityData());
          Mutation m = new Mutation(row);
          COMPACT_COL.putDelete(m);
          mutations.add(m);
        }
```

Reply:
Implemented in 69b3c48